homeassistant/components/denonavr/config_flow.py

```diff
@@ -235,11 +243,11 @@ async def async_step_ssdp(self, discovery_info):
         return await self.async_step_confirm()
 
     @staticmethod
-    def construct_unique_id(model_name, serial_number):
+    def construct_unique_id(model_name: str, serial_number: str) -> str:
         """Construct the unique id from the ssdp discovery or user_step."""
         return f"{model_name}-{serial_number}"
 
-    async def async_get_mac(self, host):
+    async def async_get_mac(self, host: str) -> Optional[str]:
```

Comment:
This is existing and doesn't need to be fixed now, but I wanted to point it out.

We don't allow usage of `getmac` to get the mac address https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#example-acceptable-sources-for-a-unique-id

Reply:
I removed the mac address and getmac completely.
There is already the `serial_number` which is read from a device description endpoint. In fact serial number and mac address share the same value.